### PR TITLE
[DOCS] Adds more detail on disk usage of kNN quantized vectors

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds more detail about the disk usage of quantized vectors to the kNN docs.

### Preview

[Reduce vector memory footprint](https://elasticsearch_bk_105724.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/tune-knn-search.html#_reduce_vector_memory_foot_print)

### CODE REVIEW

File: docs/reference/how-to/knn-search.asciidoc [hunk 1]
```diff
@@ -17,7 +17,8 @@ The default <<dense-vector-element-type,`element_type`>> is `float`. But this
 can be automatically quantized during index time through
 <<dense-vector-quantization,`quantization`>>. Quantization will reduce the
 required memory by 4x, but it will also reduce the precision of the vectors and
-increase disk usage for the field (by up to 25%).
+increase disk usage for the field (by up to 25%). Increased disk usage is a
+result of {es} storing both the quantized and the unqantized vectors.
```
Comment (leemthompo) [inline docs/reference/how-to/knn-search.asciidoc:21]:
```suggestion
result of {es} storing both the quantized and the unquantized vectors.
For example, an increase in disk usage by 10GB can result in a 30GB reduction in required memory.
```

Fix typo, and suggestion to add concrete example :)

**Review by leemthompo — APPROVED** [review]
👍

File: docs/reference/how-to/knn-search.asciidoc [hunk 2]
```diff
@@ -17,7 +17,9 @@ The default <<dense-vector-element-type,`element_type`>> is `float`. But this
 can be automatically quantized during index time through
 <<dense-vector-quantization,`quantization`>>. Quantization will reduce the
 required memory by 4x, but it will also reduce the precision of the vectors and
-increase disk usage for the field (by up to 25%).
+increase disk usage for the field (by up to 25%). Increased disk usage is a
+result of {es} storing both the quantized and the unquantized vectors.
+For example, an increase in disk usage by 10GB can result in a 30GB reduction in required memory.
```
Comment (benwtrent) [inline docs/reference/how-to/knn-search.asciidoc:22]:
```suggestion
For example, when quantizing 40GB of floating point vectors and extra 10GB of data will be stored for the quantized vectors. Totaling in a disk usage of 50GB, but the memory usage for fast search will be reduced to 10GB.
```

Comment (benwtrent) [reply docs/reference/how-to/knn-search.asciidoc:22]: I am good with that :). As long as we talk about ending totals and starting totals and not just picking `40` and `10` out of thin air.

**Review by benwtrent — APPROVED** [review]

**Comment by szabosteve** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.13|[](https://github.com/elastic/elasticsearch/pull/105743)|
|✅|8.12|[](https://github.com/elastic/elasticsearch/pull/105744)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**